File: A3/A3-026/A3-026.cpp
#include<iostream>
#include<vector>
using namespace std;

int R, C;
const char SYMBOL[4] = {'-', '+', 'x', '*'};
vector<vector<vector<int>>> img;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input images dimension
    cin >> R >> C;
    img.assign(2, vector<vector<int>>(R, vector<int>(C)));
    // Input images
    for(int i = 0; i < 2; i++)
    {
        for(int r = 0; r < R; r++)
        {
            string symbol;
            cin >> symbol;
            for(int c = 0; c < C; c++)
            {
                if(symbol[c] == '-') { img[i][r][c] = 0; }
                if(symbol[c] == '+') { img[i][r][c] = 1; }
                if(symbol[c] == 'x') { img[i][r][c] = 2; }
            }
        }
    }

    // Output overlapped images
    for(int r = 0; r < R; r++)
    {
        for(int c = 0; c < C; c++)
        {
            int s1 = img[0][r][c];
            int s2 = img[1][r][c];
            // Both image has same symbol
            if(s1 == s2) { cout << SYMBOL[s1]; }
            // Both image has different symbol
            else { cout << SYMBOL[s1 + s2]; }
        }
        cout << '\n';
    }
    return 0;
}

File: A3/A3-026/A3-026.py
# Input images dimension
R, C = map(int, input().split())

# Input images
images = []
for i in range(2):
    img = []
    for r in range(R):
        row = []
        for char in input().strip():
            if(char == '-'):
                row.append(0)
            if(char == '+'):
                row.append(1)
            if(char == 'x'):
                row.append(2)
        img.append(row)
    images.append(img)

# Output overlapped images
overlapped_img = ''
SYMBOLS = ['-', '+', 'x', '*']
for r in range(R):
    for c in range(C):
        s1 = images[0][r][c]
        s2 = images[1][r][c]
        # Both image has the same symbol
        if(s1 == s2):
            overlapped_img += SYMBOLS[s1]
        # Both image has different symbol
        else:
            overlapped_img += SYMBOLS[s1 + s2]
    overlapped_img += '\n'
print(overlapped_img)